packages/runtime-common/realm.ts

```diff
@@ -623,13 +644,19 @@ export class Realm {
     return this.#searchIndex.isIgnored(url);
   }
 
-  private async search(request: Request): Promise<Response> {
+  private async search(
+    request: Request,
+    response: Response
+  ): Promise<Response> {
     let doc = await this.#searchIndex.search(
       parseQueryString(new URL(request.url).search.slice(1)),
       { loadLinks: true }
     );
     return new Response(JSON.stringify(doc, null, 2), {
```

Comment:
this is an example of a response that should not be instantiate. There should be no `new Response`, rather the response that is used should be the one that is passed in.